### PR TITLE
test(mcp): add unit tests for broker config validation

### LINKED ISSUE #278: test: add unit tests for MCP broker config validation
Parent: #233

## Scope

Add unit tests to `filters/src/agentic/mcp/broker/config.rs` (~480 lines, zero inline test coverage).

## Key areas to test

- `build_config()` validation: valid/invalid YAML, missing required fields
- Tool schema validation: JSON Schema parsing, invalid schemas
- Unique tool name enforcement: duplicate detection across servers
- `CatalogTool` / `ToolConfig` / `McpServerConfig` construction and defaults
- Path traversal checks in server configs
- Version validation
- Boundary values for timeouts and limits

## Acceptance criteria

- `#[cfg(test)]` module added to `config.rs`
- Tests cover happy paths, validation errors, edge cases
- `make test` passes
- `make lint` passes

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Add 115+ inline unit tests for the MCP broker config module (`filters/src/agentic/mcp/broker/config.rs`)
- Cover serde defaults, `deny_unknown_fields`, `validate_schema_object` edge cases, `validate_path`, `validate_unique_server_names`, `validate_server_clusters`, `validate_server_paths`, `validate_tool_names`, `validate_unique_exposed_names`, `validate_cache_fields_for_profile`, `validate_versions`, `build_config` integration, and `build_catalog_entry`
- Use `#[cfg_attr(not(test), expect(dead_code, ...))]` for fields only read in tests

Closes #278

## Test plan

- [x] `cargo test -p praxis-ai-filters -- broker::config::tests` — all 115 tests pass
- [x] `make lint` — clippy + nightly rustfmt clean

### CODE REVIEW

File: filters/src/agentic/mcp/broker/tests.rs [hunk 1]
```diff
@@ -507,6 +509,143 @@ servers:
...
+    );
+    assert_eq!(
+        catalog[0].input_schema["type"], "object",
+        "input_schema type should be object"
+    );
+    assert_eq!(
+        catalog[0].input_schema["properties"]["city"]["type"], "string",
+        "input_schema should preserve property types"
+    );
+    // Verify routing fields via Debug to avoid triggering dead_code for
```
[the remark this answers is not in the document]
Comment (leseb) [reply filters/src/agentic/mcp/broker/tests.rs]: https://github.com/praxis-proxy/ai/pull/291/commits/58d4b2013c8721a3d5fb604e5a3889e1d179868d

File: filters/src/agentic/mcp/broker/config.rs [hunk 2]
```diff
@@ -478,3 +478,705 @@ fn build_catalog_entry(server: &McpServerConfig, tool: &ToolConfig) -> CatalogTo
...
+    clippy::unwrap_used,
+    clippy::expect_used,
+    clippy::needless_raw_strings,
+    clippy::needless_raw_string_hashes,
+    reason = "tests"
+)]
+mod tests {
+    use super::*;
+
+    // ---- CacheScope ----
```
[the remark this answers is not in the document]
Comment (leseb) [reply filters/src/agentic/mcp/broker/config.rs]: https://github.com/praxis-proxy/ai/pull/291/commits/58d4b2013c8721a3d5fb604e5a3889e1d179868d

**Review by aslakknutsen — APPROVED** [review]
